maa.jwt.verifier.sevsnp/readme.md

```diff
@@ -0,0 +1,104 @@
+# SEV-SNP Attestation Sample with Azure Attestation
+
+This is a minimal .NET console application that interacts with the Microsoft Azure Attestation (MAA) service for SEV-SNP attestation type.
+
+## Remote Attestation Flow with SEV-SNP
+
+1. Collect Attestation Evidence. The confidential environment undergoing attestation (e.g., a confidential ACI container, SGX enclave, or Azure Confidential VM) generates attestation evidence that reflects its identity, state, and configuration.
+
+2. Submit Evidence for Attestation. The attestation evidence is sent to the **Microsoft Azure Attestation (MAA)** service, which verifies the evidence against platform-specific security guarantees. If the attestation succeeds, MAA issues a signed JWT token.
+
+3. Validate the Attestation Token. The signed JWT token is provided to a **relying party** that requires trust verification before interacting with the confidential environment (e.g., **Azure Key Vault for secure key release**). The relying party validates the JWT token by verifying signatures and inspecting the attestation evidence to confirm MAA's TEE hosting environment.
+
+4. Authorization Decision. Based on the validated attestation claims, the relying party makes an authorization decision on whether to trust and engage with the attested environment.
+
+## How to Verify MAA JWT for SEV-SNP
+
+### Verify that the attestation token is generated by Azure Attestation
+
+Attestation tokens from Azure Attestation are JWTs signed using a certificate chain rooted in a trusted MAA certificate. The token header includes a `jku` (JWK Set URL) parameter, which points to the endpoint hosting the signing certificates. The verifier must retrieve the signing keys from this URL, select the key matching the `kid` in the JWT header, and validate the token's signature and expiration.
+
+The signing certificates are also available via the [OpenID metadata endpoint](https://learn.microsoft.com/en-us/rest/api/attestation/metadata-configuration/get?view=rest-attestation-2022-08-01&tabs=HTTP#get-openid-metadata), for example:  
+`https://<your-instance>.attest.azure.net/.well-known/openid-configuration`
+
+See [code](Program.cs#L161) for an implementation example.
+
+### Verify that the certificate is rooted in the trusted MAA root key
```

Comment:
This section should be removed.  What we do in the remainder of the steps is to verify that the key in the self-signed MAA cert that signed the JWT is rooted in SEVSNP attestation evidence chaining back to AMD and PRSS-ACI and MAA CCE policy.